### PR TITLE
Marshal errors with *ErrorReturn type

### PR DESCRIPTION
**Pull request by whaught**
Fixes https://github.com/google/exposure-notifications-verification-server/issues/1581

## Proposed Changes

* Marshal json with ErrorReturn struct rather than the IssueResponse struct, although identical they differ in that ErrorReturn does not include `OmitEmpty`
* Add an error code `sms_failure` for generic Twilio errors

### CODE REVIEW

File: pkg/controller/issueapi/handle_issue.go [hunk 1]
```diff
@@ -104,16 +104,16 @@ func (c *Controller) decodeAndIssue(ctx context.Context, w http.ResponseWriter,
...
-	resp := res.IssueCodeResponse()
-	if resp.Error != "" {
-		if result.HTTPCode == http.StatusInternalServerError {
-			controller.InternalError(w, r, c.h, errors.New(resp.Error))
-			return
-		}
-		c.h.RenderJSON(w, result.HTTPCode, resp)
+
+	switch res.HTTPCode {
+	case http.StatusInternalServerError:
```
Comment (sethvargo) [inline pkg/controller/issueapi/handle_issue.go:109]: I'm not sure I follow why these cases behave differently?

**Comment by mikehelmick** [comment]
can we add tests?

**Comment by whaught** [comment]
> can we add tests?

That's actually difficult to do - we'd need a known case where there is no error code to assert that the key is there with empty-string after json marshal. In this case I've added a code for twilio failure so that's covered now.

File: pkg/controller/issueapi/handle_issue.go (hunk 1 above)
Comment (whaught) [reply pkg/controller/issueapi/handle_issue.go:109]: before I was rendering `struct IssueCodeResponse` which has:
`ErrorCode string 'json:"errorCode,omitempty"'`

We put error and errorCode on the responses which perhaps help document the possible responses. In practice we respond with:
`type ErrorReturn struct`
which has
`ErrorCode string 'json:"errorCode"'` with no `omitempty`

Comment (whaught) [reply pkg/controller/issueapi/handle_issue.go:109]: the key difference here (other than the switch which I just thought looked better:
`c.h.RenderJSON(w, result.HTTPCode, resp)`
is now
`c.h.RenderJSON(w, result.HTTPCode, res.ErrorReturn)`

**Comment by mikehelmick** [comment]
this causes the issue_tests to timeout.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**